### PR TITLE
milestone-5 event-handlers

### PR DESCRIPTION
**Pull request by sz16900**
- add click handler events to the app, button and button panel
- final touch-ups
- final wiring
- catch division by zero error

### CODE REVIEW

**Review by Forison — CHANGES_REQUESTED** [review]
## Status:: Changes required❌.
# 🙋‍♂️ Seth
#### You have done a great job on this one, your project meets most the technical requirements, but there are a few functional issues we have to address to make the application user friendly and output correct info to the user🙍‍♂️.
* [x] I am unable to perform decimal computation ie ```0.4 + 0.4  ```
* [x] The styling should enable the user to see all the inputted value even the exceed the length of the display screen. 
Check the image below👇
overflow
Make the digit break or shrink size when the length increases.
* [x] Use Object destructuring to efficiently obtain your data instead of  ```const copyObject = this.state; ```
 in your App.js
* [x] Let use ES6 syntax to avoid having to bind methods we create 
```this.handleClick = this.handleClick.bind(this); ```
We can define ```handleClick = () => { }``` this way to avoid binding.

**[OPTIONAL REQUIREMENT ]**

* [ ] It will more user friendly to have the current operation selected appear somewhere on the screen so the user can see what is happening.

### That is it kindly make these changes and submit a re-review👨‍🔧.

#### Happy coding👨‍💻

Addo Forison

File: src/components/App.js [hunk 1]
```diff
@@ -3,12 +3,36 @@ import React from 'react';
...
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  handleClick = buttonName => {
+    const { ...copyState } = this.state;
+    const { ...calcResult } = calculate(copyState, buttonName);
+    this.setState(calcResult);
```
Comment (suretrust) [inline src/components/App.js]: You can use React's `prevState` for this. You could read more about it [here](https://stackoverflow.com/questions/54807454/what-is-prevstate-in-reactjs). This is completely optional, just pointing it out for future reference.

```suggestion
    this.setState(prevState => calculate(prevState, buttonName))
```

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi @sz16900, 👋 

### Great job implementing some of the suggested changes! 🙌 Please see my comments. 👇 

- [ ]  I am still unable to perform decimal computation ie `0.4` + `0.4`. `0.anything` is not even pressing.

**KIndly submit another CRR once these issues have been solved.** _Happy coding!_ 🎉

**Review by shshamim63 — APPROVED** [review]
## :100: Approved
___
Hi @sz16900 :wave: ,
Congratulation, you have successfully resolve the bug, now it is working properly. Nothing is blocking me from approving this project. Great job :clap: . You can go ahead and merge it.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Pull request merged**